package/gluon-xlat464-clat/files/xlat464clat.sh

```diff
@@ -0,0 +1,173 @@
+  config_get		remote_style	"${config}" "remote_style"
+  config_get		remote_v4	"${config}" "remote_v4"
+  config_get		remote_v6	"${config}" "remote_v6"
+  config_get		remote_ea_len	"${config}" "remote_ea_len"
+  config_get		remote_psid	"${config}" "remote_psid_offset"
+
+
+  zone="${zone:-wan}"
+  zone4="${zone4:-lan}"
+  local_v6="${local_v6:-fd00:13:37:13:37::/96}"
+  remote_v4="${remote_v4:-0.0.0.0/0}"
+
+  [ $debug -ne 0 ] && clat_cfg="$clat_cfg debug 1"
+  [ ${local_fmr} -ne 0 ] && clat_cfg="$clat_cfg local.fmr-flag 1"
+
+  clat_cfg="${clat_cfg} local.style ${local_style:-RFC6052} local.v4 ${local_v4:-192.168.1.0/24} local.v6 ${local_v6}"
+  clat_cfg="${clat_cfg} local.ea-len ${local_ea_len:-0} local.psid-offset ${local_psid:-0}"
+  clat_cfg="${clat_cfg} remote.style ${remote_style:-RFC6052} remote.v4 ${remote_v4} remote.v6 ${remote_v6:-64:ff9b::/96}"
+  clat_cfg="${clat_cfg} remote.ea-len ${remote_ea_len:-0} remote.psid-offset ${remote_psid:-0}"
+
+  # create nat46 interface
+  echo "add ${config}" > ${CTR}
+
+  # configure nat46
+  echo "${clat_cfg}" > ${CTR}
+
+  [ "${ip4table}" ] && ip -4 rule add from ${local_v4} lookup ${ip4table}
+
+  proto_init_update "${config}" 1
+
+	  # add routes
+	  case "${local_v6}" in
+	    *:*/*)
+	      proto_add_ipv6_route "${local_v6%%/*}" "${local_v6##*/}"
+	    ;;
+	    *:*)
+	      proto_add_ipv6_route "${local_v6%%/*}" "128"
+	    ;;
+	  esac
+
+	  case "${remote_v4}" in
+	    *.*/*)
+	      proto_add_ipv4_route "${remote_v4%%/*}" "${remote_v4##*/}" "" "" 2048
+	    ;;
+	    *.*)
+	      proto_add_ipv4_route "${remote_v4%%/*}" "32" "" "" 2048
+	    ;;
+	  esac
+
+	proto_add_data
+	[ "${zone}" != "-" ] && json_add_string zone "${zone}"
+
+	json_add_array firewall
+	  # if forwarding within "zone" and between zone<->zone4 is allowed you can set zone = "-"
+	  if [ "${zone}" != "-" ]; then
+	  	json_add_object ""
+	  		json_add_string type rule
+	  		json_add_string family inet6
+	  		json_add_string proto all
+	  		json_add_string direction in
+			json_add_string dest "${zone}"
+			json_add_string src "${zone}"
+	  		json_add_string src_ip "$local_v6"
+	  		json_add_string target ACCEPT
+	  	json_close_object
+		# if a forwarding between zone and zone4 exists (e.g. lan<->wan) you can set zone4 = "-"
+	  	if [ "${zone4}" != "-" ]; then
+		  	json_add_object ""
+		  		json_add_string type rule
+		  		json_add_string family inet
+		  		json_add_string proto all
+		  		json_add_string direction out
+				json_add_string dest "${zone}"
+				[ "$remote_v4" != "0.0.0.0/0" ] && json_add_string dest_ip "$remote_v4"
+				json_add_string src "${zone4}"
+		  		json_add_string src_ip "$local_v4"
+		  		json_add_string target ACCEPT
+		  	json_close_object
+	  	fi
+	  fi
+
+	json_close_array
+
+	proto_close_data
+
+  proto_send_update "$config"
+}
+
+
+proto_xlat464clat_teardown() {
+  local config="$1"
+  echo "del ${config}" > ${CTR}
+}
+
+
+[ -n "$INCLUDE_ONLY" ] || {
+  add_protocol xlat464clat
+}
+
```

Comment:
empty line at EOF - and many double empty lines in the script, some may be removed?